llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2095,15 +2095,24 @@ LogicalResult WinogradInputTransformOp::verify() {
   }
   auto inputType = input().getType().cast<ShapedType>();
   auto outputType = output().getType().cast<ShapedType>();
-  ArrayRef<int64_t> inputShape = inputType.getShape();
-  if (inputShape.size() != 4) {
-    return op->emitOpError("expected input operand to have rank 4");
-  }
-  ArrayRef<int64_t> outputShape = outputType.getShape();
   if (outputType.getElementType() != inputType.getElementType()) {
     return op->emitOpError(
         "expected input/output element types to be identical");
   }
+  unsigned inputRank = inputType.getRank();
+  unsigned outputRank = outputType.getRank();
+  
+  if (inputRank != 2 && inputRank != 4) {
+    return op->emitOpError("expected input operand to have rank either 2 or 4");
+  }
+  
+  if (inputRank == 2) {
```

Comment:
you can make this a single condition, `inputRank == 2 && outputRank != 2`

Reply:
That's correct but I want to make an early exit in the verification process on `inputRank == 2` - hence I chose to write it this way.

I need to fix a few bugs in this + fix lit tests + add lit tests for `onlyTile` - have added WinogradOutputTransformOp in the latest push.